gem-changelog-update-check/action.yml

```diff
@@ -0,0 +1,29 @@
+name: Check if CHANGELOG needs updating
+description: Checks if files in lib/ have changed; if so, check if CHANGELOG.md has been updated
+author: Workflow Improvement Team
+inputs:
+  changelog-file-path:
+    description: Relative path to the CHANGELOG file
+    required: false
+    default: CHANGELOG.md
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Check if CHANGELOG needs updating
+      run: |
+        file_regex='^lib/|^bin/|^spec/|\.gemspec$'
+        changes="$(git diff --name-only --diff-filter=ACMRT origin/main HEAD)"
+        changelog_file="${{ inputs.changelog-file-path }}"
+        echo "Changes relative to main branch:"
+        echo "$changes"
+        if [ -f "$changelog_file" ]; then
+        	echo "$changes" | grep -q -E "$file_regex" && \
+        	echo "$changes" | grep -q "$changelog_file"
```

Comment:
Use `fgrep -qx` or `grep -Fqx` to search for the exact line "CHANGELOG.md" (or whatever was specified).